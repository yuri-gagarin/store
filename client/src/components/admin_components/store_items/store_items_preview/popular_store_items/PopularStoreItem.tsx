import React from "react";
import { Button, Item } from "semantic-ui-react";
// helpers //
import { ConvertDate, setDefaultImage } from "../../../../helpers/displayHelpers";

interface Props {
  popularStoreItem: IStoreItemData;
}

const PopularStoreItem: React.FC<Props> = ({ popularStoreItem }): JSX.Element => {
  const { name, images, price, createdAt } = popularStoreItem;
  
  return (
    <Item>
      <Item.Image size="tiny" src={setDefaultImage(images)} />
      <Item.Content>
        <Item.Header>{name}</Item.Header>
        <Item.Meta>
          <span>Priced At: <strong>{price}</strong></span>
          <span>-</span>
          <span>Posted: <strong>{ConvertDate.international(createdAt)}</strong></span>
        </Item.Meta>
        <Button basic color="blue" content="More..." />
      </Item.Content>

    </Item>
  );
};

export default PopularStoreItem;
