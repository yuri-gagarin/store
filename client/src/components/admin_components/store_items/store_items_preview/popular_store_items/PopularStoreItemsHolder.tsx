import React from "react";
import { Item } from "semantic-ui-react";
// additional componets //
import PopularStoreItem from "./PopularStoreItem";
// css imports //
import "./css/popularStoreItemsHolder.css";

interface Props {
  popularStoreItems: IStoreItemData[];
  // loading: boolean;
}

const PopularStoreItemHolder: React.FC<Props> = ({ popularStoreItems }): JSX.Element => {
  return (
    <div className="popStoreItemsHolder">
      <div className="popStoreItemsHolTitle">Popular Store Items</div>
      <Item.Group divided>
        {
          popularStoreItems.map((storeItem) => {
            return (
              <PopularStoreItem key={storeItem._id}  popularStoreItem={storeItem} />
            );
          })
        }
      </Item.Group>
    </div>  
  );
};

export default PopularStoreItemHolder;