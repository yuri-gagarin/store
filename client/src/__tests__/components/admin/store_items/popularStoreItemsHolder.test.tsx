import React from "react";
import PopularStoreItemsHolder from "../../../../components/admin_components/store_items/store_items_preview/popular_store_items/PopularStoreItemsHolder";
// test dependencies //
import { mount, ReactWrapper } from "enzyme";
import PopularStoreItem from "../../../../components/admin_components/store_items/store_items_preview/popular_store_items/PopularStoreItem";

describe("'PopularStoreItemsHolder' component render tests", () => {
  let mockStore: IStoreData; let mockStoreItems: IStoreItemData[];
  const date = new Date("1/1/2019").toString();
  let wrapper: ReactWrapper;

  beforeAll(() => {
    mockStore = {
      _id: "1",
      title: "title",
      description: "description",
      images: [],
      createdAt: date
    };
    mockStoreItems = [
      {
        _id: "111",
        storeId: mockStore._id,
        storeName: mockStore.title,
        name: "name",
        price: "100",
        description: "description",
        details: "details",
        categories: [ "sports", "outdoors" ],
        images: [],
        createdAt: date
      },
      {
        _id: "112",
        storeId: mockStore._id,
        storeName: mockStore.title,
        name: "name",
        price: "100",
        description: "description",
        details: "details",
        categories: [ "sports" ],
        images: [],
        createdAt: date
      },
      {
        _id: "113",
        storeId: mockStore._id,
        storeName: mockStore.title,
        name: "name",
        price: "100",
        description: "description",
        details: "details",
        categories: [ "sports", "outdoors", "education" ],
        images: [],
        createdAt: date
      }
    ];

    wrapper = mount(
      <PopularStoreItemsHolder popularStoreItems={mockStoreItems} />
    );
  });

  describe("Rendering in its default state", () => {
    it("Should correctly render and match snapshot", () => {
      expect(wrapper.find(PopularStoreItemsHolder)).toMatchSnapshot();
    });
    it("Should render a correct number of 'PopularStoreItem' components", () => {
      const popularStoreItems = wrapper.find(PopularStoreItem);
      expect(popularStoreItems.length).toEqual(mockStoreItems.length);
    });
  });

});