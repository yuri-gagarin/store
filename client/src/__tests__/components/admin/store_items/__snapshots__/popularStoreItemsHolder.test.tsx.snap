// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'PopularStoreItemsHolder' component render tests Rendering in its default state Should correctly render and match snapshot 1`] = `
<PopularStoreItemHolder
  popularStoreItems={
    Array [
      Object {
        "_id": "111",
        "categories": Array [
          "sports",
          "outdoors",
        ],
        "createdAt": "Tue Jan 01 2019 00:00:00 GMT+0200 (Eastern European Standard Time)",
        "description": "description",
        "details": "details",
        "images": Array [],
        "name": "name",
        "price": "100",
        "storeId": "1",
        "storeName": "title",
      },
      Object {
        "_id": "112",
        "categories": Array [
          "sports",
        ],
        "createdAt": "Tue Jan 01 2019 00:00:00 GMT+0200 (Eastern European Standard Time)",
        "description": "description",
        "details": "details",
        "images": Array [],
        "name": "name",
        "price": "100",
        "storeId": "1",
        "storeName": "title",
      },
      Object {
        "_id": "113",
        "categories": Array [
          "sports",
          "outdoors",
          "education",
        ],
        "createdAt": "Tue Jan 01 2019 00:00:00 GMT+0200 (Eastern European Standard Time)",
        "description": "description",
        "details": "details",
        "images": Array [],
        "name": "name",
        "price": "100",
        "storeId": "1",
        "storeName": "title",
      },
    ]
  }
>
  <div
    className="popStoreItemsHolder"
  >
    <div
      className="popStoreItemsHolTitle"
    >
      Popular Store Items
    </div>
    <ItemGroup
      divided={true}
    >
      <div
        className="ui divided items"
      >
        <PopularStoreItem
          key="111"
          popularStoreItem={
            Object {
              "_id": "111",
              "categories": Array [
                "sports",
                "outdoors",
              ],
              "createdAt": "Tue Jan 01 2019 00:00:00 GMT+0200 (Eastern European Standard Time)",
              "description": "description",
              "details": "details",
              "images": Array [],
              "name": "name",
              "price": "100",
              "storeId": "1",
              "storeName": "title",
            }
          }
        >
          <Item>
            <div
              className="item"
            >
              <ItemImage
                size="tiny"
                src="https://react.semantic-ui.com/images/wireframe/image.png"
              >
                <Image
                  as="img"
                  size="tiny"
                  src="https://react.semantic-ui.com/images/wireframe/image.png"
                  ui={true}
                  wrapped={true}
                >
                  <div
                    className="ui tiny image"
                  >
                    <img
                      src="https://react.semantic-ui.com/images/wireframe/image.png"
                    />
                  </div>
                </Image>
              </ItemImage>
              <ItemContent>
                <div
                  className="content"
                >
                  <ItemHeader>
                    <div
                      className="header"
                    >
                      name
                    </div>
                  </ItemHeader>
                  <ItemMeta>
                    <div
                      className="meta"
                    >
                      <span>
                        Priced At: 
                        <strong>
                          100
                        </strong>
                      </span>
                      <span>
                        -
                      </span>
                      <span>
                        Posted: 
                        <strong>
                          2019-Jan-1
                        </strong>
                      </span>
                    </div>
                  </ItemMeta>
                  <Button
                    as="button"
                    basic={true}
                    color="blue"
                    content="More..."
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui blue basic button"
                          >
                            More...
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui blue basic button"
                            >
                              More...
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui blue basic button"
                          onClick={[Function]}
                        >
                          More...
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                </div>
              </ItemContent>
            </div>
          </Item>
        </PopularStoreItem>
        <PopularStoreItem
          key="112"
          popularStoreItem={
            Object {
              "_id": "112",
              "categories": Array [
                "sports",
              ],
              "createdAt": "Tue Jan 01 2019 00:00:00 GMT+0200 (Eastern European Standard Time)",
              "description": "description",
              "details": "details",
              "images": Array [],
              "name": "name",
              "price": "100",
              "storeId": "1",
              "storeName": "title",
            }
          }
        >
          <Item>
            <div
              className="item"
            >
              <ItemImage
                size="tiny"
                src="https://react.semantic-ui.com/images/wireframe/image.png"
              >
                <Image
                  as="img"
                  size="tiny"
                  src="https://react.semantic-ui.com/images/wireframe/image.png"
                  ui={true}
                  wrapped={true}
                >
                  <div
                    className="ui tiny image"
                  >
                    <img
                      src="https://react.semantic-ui.com/images/wireframe/image.png"
                    />
                  </div>
                </Image>
              </ItemImage>
              <ItemContent>
                <div
                  className="content"
                >
                  <ItemHeader>
                    <div
                      className="header"
                    >
                      name
                    </div>
                  </ItemHeader>
                  <ItemMeta>
                    <div
                      className="meta"
                    >
                      <span>
                        Priced At: 
                        <strong>
                          100
                        </strong>
                      </span>
                      <span>
                        -
                      </span>
                      <span>
                        Posted: 
                        <strong>
                          2019-Jan-1
                        </strong>
                      </span>
                    </div>
                  </ItemMeta>
                  <Button
                    as="button"
                    basic={true}
                    color="blue"
                    content="More..."
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui blue basic button"
                          >
                            More...
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui blue basic button"
                            >
                              More...
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui blue basic button"
                          onClick={[Function]}
                        >
                          More...
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                </div>
              </ItemContent>
            </div>
          </Item>
        </PopularStoreItem>
        <PopularStoreItem
          key="113"
          popularStoreItem={
            Object {
              "_id": "113",
              "categories": Array [
                "sports",
                "outdoors",
                "education",
              ],
              "createdAt": "Tue Jan 01 2019 00:00:00 GMT+0200 (Eastern European Standard Time)",
              "description": "description",
              "details": "details",
              "images": Array [],
              "name": "name",
              "price": "100",
              "storeId": "1",
              "storeName": "title",
            }
          }
        >
          <Item>
            <div
              className="item"
            >
              <ItemImage
                size="tiny"
                src="https://react.semantic-ui.com/images/wireframe/image.png"
              >
                <Image
                  as="img"
                  size="tiny"
                  src="https://react.semantic-ui.com/images/wireframe/image.png"
                  ui={true}
                  wrapped={true}
                >
                  <div
                    className="ui tiny image"
                  >
                    <img
                      src="https://react.semantic-ui.com/images/wireframe/image.png"
                    />
                  </div>
                </Image>
              </ItemImage>
              <ItemContent>
                <div
                  className="content"
                >
                  <ItemHeader>
                    <div
                      className="header"
                    >
                      name
                    </div>
                  </ItemHeader>
                  <ItemMeta>
                    <div
                      className="meta"
                    >
                      <span>
                        Priced At: 
                        <strong>
                          100
                        </strong>
                      </span>
                      <span>
                        -
                      </span>
                      <span>
                        Posted: 
                        <strong>
                          2019-Jan-1
                        </strong>
                      </span>
                    </div>
                  </ItemMeta>
                  <Button
                    as="button"
                    basic={true}
                    color="blue"
                    content="More..."
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui blue basic button"
                          >
                            More...
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui blue basic button"
                            >
                              More...
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui blue basic button"
                          onClick={[Function]}
                        >
                          More...
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                </div>
              </ItemContent>
            </div>
          </Item>
        </PopularStoreItem>
      </div>
    </ItemGroup>
  </div>
</PopularStoreItemHolder>
`;
